fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Response.kt

```diff
@@ -12,7 +16,14 @@ class Response {
     var httpContentLength = 0L
 
     //data
-    var data = ByteArray(0)
+    internal var dataStream: InputStream = ByteArrayInputStream(ByteArray(0))
```

Comment:
Is it intentional that `dataStream` is `internal`, and that `data` was switched from `var` to `val`?

This makes it really hard to override the built in HttpClient, and therefore also very hard to stub out the actual web requests when unittesting.

If `dataStream` was not `internal`, or `data` was a `var`, I would be able to solve #186